core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -103,34 +122,45 @@ public void close() throws IOException {
      */
     public void abort() throws IOException {
         close();
-        tmpFile.delete();
+        Files.deleteIfExists(tmpFile);
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
-            try {
-                Util.deleteFile(destFile);
-            } catch (IOException x) {
-                tmpFile.delete();
-                throw x;
-            }
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpFile, destFile, StandardCopyOption.ATOMIC_MOVE);
+        } catch (IOException e) {
```

Comment:
@jglick no. If the file exists a generic IOException can be thrown iff atomic move is suppported and the implementation does not overwrite and it does not implement the *optional* FileAlreadyExistsException